numpy/core/src/multiarray/conversion_utils.c

```diff
@@ -533,59 +546,34 @@ PyArray_SearchsideConverter(PyObject *obj, void *addr)
 NPY_NO_EXPORT int
 PyArray_OrderConverter(PyObject *object, NPY_ORDER *val)
 {
-    char *str;
-    /* Leave the desired default from the caller for NULL/Py_None */
-    if (object == NULL || object == Py_None) {
+    /* Leave the desired default from the caller for Py_None */
+    if (object == Py_None) {
```

Comment:
will skipping the NULL check cause any issue fo C API users?

Reply:
No reasonable c API user should be passing `NULL`.
Probably worth checking to see if we've set a precedent with other converter functions.

Reply:
Yeah, was also a bit confused for a second. Initializing `NPY_ORDER` to anything you like is fine, but passing `NULL` really is very weird. It is also very unlikely, since in most cases these type of functions should be just passed into the Python machinery (users rarely call them directly).
That said, yes, someone could in theory segfault here, but at least that is a loud error, so I do not care much either way.

Reply:
with the previous code and NULL argument this returns NPY_SUCCEED and i think now it will return NPY_FAIL. Is this not considered a breaking change, or are we saying that its not going to impact any users probably, so its fine to do it ?

Reply:
The typical use of these functions (both by us and in downstream code) is something like `PyArg_ParseTuple(args, "O&", PyArray_OrderConverter, &order))`.
For `object` to be `NULL`, `args` would have to be a tuple containing `NULL` elements, which is not a legal tuple object.

Another question is whether we call it `PyArray_OrderConverter` directly internally. We do, in exactly one place in `array_einsum`, but again it's guaranteed that `object` is non-null.

So the only case that would actually matter is users using `PyArray_OrderConverter` directly. The default should be to assume that any `PyObject *` argument must be non-null unless explicitly documented otherwise. We don't document that, so I think this change is safe.

Reply:
thanks for the explanation @eric-wieser !

> The default should be to assume that any PyObject * argument must be non-null unless explicitly documented otherwise. 

Is this documented somewhere in python c api docs/numpy c api docs or is it a standard practice that numpy and python C API users follow ?